src/layouts/Login/components/LoginForm.tsx

```diff
@@ -0,0 +1,64 @@
+import { FormControl } from "@chakra-ui/react"
+import {
+  Button,
+  FormLabel,
+  Input,
+  FormErrorMessage,
+} from "@opengovsg/design-system-react"
+import { useEffect } from "react"
+import { useForm } from "react-hook-form"
+
+export type LoginProps = {
+  email: string
+}
+
+interface LoginFormProps {
+  onSubmit: (inputs: LoginProps) => Promise<void>
+  errorMessage: string
+}
+
+const validateEmail = (value: string) =>
+  value.length > 0 || "Please enter a valid email."
+
+export const LoginForm = ({
+  onSubmit,
+  errorMessage,
+}: LoginFormProps): JSX.Element => {
+  const { handleSubmit, register, formState, setError } = useForm<LoginProps>({
+    mode: "onBlur",
+  })
+  useEffect(() => {
+    if (errorMessage)
+      setError("email", {
+        type: "server",
+        message: errorMessage,
+      })
+  }, [errorMessage, setError])
```

Comment:
Hmm separating it out into separate variables like this seems to cause issues with the error message not being cleared  - I think this is because errorMessage doesn't actually change until we call the endpoint again, but we actually want to clear this message as soon as the user edits the field. To get this to work seems like it'll make the code harder to read than the current implementation, wdyt?

Reply:
yea you're right! thanks for testing this out!